### PR TITLE
restructure documentation and add common tech

### PR DESCRIPTION
**Pull request by denisok**
This PR restructures development documentation and moves Common Tech article from Confluence to GitHub

### CODE REVIEW

File: docs/process/common_tech_lib_approaches.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+* [logrus](https://github.com/sirupsen/logrus) or stdlib `log` package should be used for logging. Always log to unbuffered stderr, let process supervisor do the rest.
+* [prometheus client](https://github.com/prometheus/client_golang) is used for exposing internal metrics of application and gRPC library.
+* [testify](https://github.com/stretchr/testify) or stdlib `testing` package should be used for writing tests. Testify should be used only for `assert` and `require` packages – suites here have some problems with logging and parallel tests. Common setups and teardowns should be implemented with `testing` [subtests](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks).
+* [golangci-lint](https://github.com/golangci/golangci-lint) is used for static code checks.
+* [gocov.io](http://gocov.io/) and [gocoverutil](https://github.com/AlekSi/gocoverutil) gather code coverage metrics.
+* [Docker Compose](https://docs.docker.com/compose/) is used for a local development environment and in CI.
+* [Kong](https://github.com/alecthomas/kong) for CLI.
```
Comment (BupycHuk) [inline docs/process/common_tech_lib_approaches.md:22]: let's mention that we use kingpin for exporters and some other components

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -158,7 +158,7 @@ Please see [readme](https://github.com/percona/pmm-qa#readme) for details on how
 
 ## Submitting a Pull Request
 
-See [Working with Git and GitHub](GIT_AND_GITHUB.md)
+See [Working with Git and GitHub](dosc/process/GIT_AND_GITHUB.md)
```
Comment (atymchuk) [inline CONTRIBUTING.md:161]:
```suggestion
See [Working with Git and GitHub](docs/process/GIT_AND_GITHUB.md)
```

**Review by atymchuk — CHANGES_REQUESTED** [review]
This move entails a change on this line as well https://github.com/percona/pmm/pull/973/files#diff-7871fa1d10e6bf511d9cfda5a6874d1a75acb950fb1bc9fee6057eff13257521R35

File: docs/process/common_tech_lib_approaches.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  * The idea here that missing external dependency and already used port during startup is either a configuration or environment problem, or a programming bug. Neither can be fixed by simply continuing. Attempts to make startup "smart" (re-read configuration file, re-parse command-line flags, etc.) significantly complicate it. On the other hand, when an error happens during normal program execution, it typically has a proper way to communicate it to the user so they can retry.
+  * Supervisor-level dependencies between services are not strictly required (due to restarts everything will work eventually) but are nice to have: they help to avoid scarry-looking errors in logs.
+  * The big exception here is exporters: we should follow practices established by Prometheus team and not fail if the system under monitoring is not available. Other startup errors like missing certificate file should still terminate exporter.
+
+## Code style
+
+* `gofmt -s ` (note the `-s`)
```
Comment (BupycHuk) [inline docs/process/common_tech_lib_approaches.md:49]: we actually use `gofumpt`, also I think we should mention `make format` here

**Comment by denisok** [comment]
> This move entails a change on this line as well https://github.com/percona/pmm/pull/973/files#diff-7871fa1d10e6bf511d9cfda5a6874d1a75acb950fb1bc9fee6057eff13257521R35

done

**Comment by BupycHuk** [comment]
Can we add `docs/*` into paths-ignore in other workflows?

**Comment by denisok** [comment]
> Can we add `docs/*` into paths-ignore in other workflows?

done

File: docs/process/common_tech_lib_approaches.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (BupycHuk) [inline docs/process/common_tech_lib_approaches.md:49]:
```suggestion
* `gofumt -s ` (note the `-s`)
* `make format`
```

**Review by BupycHuk — APPROVED** [review]

**Review by BupycHuk — DISMISSED** [review]

**Review by atymchuk — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**